ipc/observability/src/tracing_layers.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use std::ops::Deref;
+use tracing::{field::Visit, Event, Subscriber};
+use tracing_subscriber::Registry;
+use tracing_subscriber::{layer::Context, Layer};
+
+const DOMAIN_FIELD: &str = "domain";
+const EVENT_FIELD: &str = "event";
+
+pub struct DomainEventFilterLayer<L> {
+    domains: Option<Vec<String>>,
+    events: Option<Vec<String>>,
+    inner: L,
+}
+
+impl Deref for DomainEventFilterLayer<Registry> {
```

Comment:
I feel a `.register` getter is more suited for this, this is not supposed to be a smart pointer right?

Reply:
yes good point.